Add page on init container resource calc (#30456)
DataDog/documentation: content/en/tracing/guide/init_resource_calc.md

---

> title: Init container resource usage

Reviewer: feel free to adjust description as needed but was mainly targeting the sentence capitalization here

```suggestion
title: Init Container Resource Usage
Learn how Datadog Agent v7.60+ automatically manages resource allocation for init containers that set up tracing, ensuring reliable tracer startup without impacting pod scheduling.
```

disable_toc: false
---

### Overview

> Starting in Agent [v7.60+][1], Datadog uses dynamic resource calculation for init containers that inject tracing libraries. Instead of fixed values, the init containers temporarily request all available CPU and memory for the pod, without affecting how the pod is scheduled. (Prior v7.60, init containers used conservative defaults: `50m` for CPU and `20Mi` for memory.)

Reviewer: ```suggestion
Starting in Agent [v7.60+][1], Datadog uses dynamic resource calculation for init containers that inject tracing libraries. Instead of using fixed values, the init containers temporarily request all available CPU and memory for the pod, without affecting how the pod is scheduled. (Prior to v7.60, init containers used conservative defaults: `50m` for CPU and `20Mi` for memory.)
```

> This behavior improves tracer startup reliability while respecting Kubernetes scheduling rules. Init containers run sequentially and exit before application containers start, so they don't compete for runtime resources.

Reviewer: ```suggestion
This behavior improves tracer startup reliability while respecting Kubernetes scheduling rules. Since init containers run sequentially and exit before application containers start, they don't compete for runtime resources.
```

### Pod scheduling

Kubernetes schedules pods using a formula that accounts for init containers:

<div style="text-align:center">
<pre><code>
Effective CPU/Memory request =
  max(sum of requests from all regular containers,
      max request of any single init container)
</code></pre>
</div>

> Because init containers run before application containers (and don't overlap), they can temporarily use more resources without increasing the pod's effective request, as long as no single init container requests more than the pod can tolerate.

Reviewer: ```suggestion
Since init containers run before application containers (and don't overlap with them), they can temporarily use more resources without increasing the pod's effective request. This works as long as no single init container requests more resources than the pod can tolerate.
```

### Override default behavior

> If needed, you can override init container resource usage by setting the following environment variables in the Cluster Agent configuration:

Reviewer: ```suggestion
If needed, you can override the default init container resource usage by setting the following environment variables in the Cluster Agent configuration:
```
